added usefull rubocop config. Updated all files with the rules

fanantoxa opened the pull request:
@soulcutter I've added config for rubocop and updated all files. Please merge it if it fit to you.
If it not please let me know which rules you want to on/off and I'll update PR

soulcutter commented on lib/saxerator/adapters/ox.rb:
```diff
@@ -54,7 +52,7 @@ def text(str)
       end
 
-      alias :cdata :text
+      alias cdata text
```
ha, this one always bites me.

soulcutter commented on lib/saxerator/builder/hash_builder.rb:
```diff
@@ -60,12 +60,13 @@ def add_to_hash_element(hash, name, element)
         return to_s if @text
-        return to_hash if @children.count > 0 || (@attributes.count > 0 && @config.put_attributes_in_hash?)
+        return to_hash if @children.count > 0 ||
+                          (@attributes.count > 0 && @config.put_attributes_in_hash?)
```
I strongly dislike whitespace to align things into columns. I'd rather extract methods or variables to keep this short enough to not have to wrap. Any cop that promotes that style should be disabled.

soulcutter commented on lib/saxerator/configuration.rb:
```diff
@@ -30,23 +26,23 @@ def generate_key_for(val)
 
     def hash_key_normalizer
-      @hash_key_normalizer ||= lambda { |x| x.to_s }
+      @hash_key_normalizer ||= -> (x) { x.to_s }
```
The current build matrix includes Ruby 1.9.3 which does not have stabby lambdas.

Honestly, though, I'm ok dropping 1.9.3 support. It's a dead ruby at this point.

soulcutter commented on lib/saxerator/configuration.rb:
```diff
@@ -69,7 +65,8 @@ def put_attributes_in_hash?
       if @output_type != :hash && @put_attributes_in_hash
-        raise ArgumentError.new("put_attributes_in_hash! is only valid when using output_type = :hash (the default)'")
+        raise ArgumentError, "put_attributes_in_hash! is only valid \
+                              when using output_type = :hash (the default)'"
```
Prefer not to indent into columns here.

fanantoxa replied:
@soulcutter yes, but it's really hard to see this line.

soulcutter commented on lib/saxerator/latches/child_of.rb:
```diff
@@ -6,20 +6,15 @@ def initialize(name)
-          @depths.push 1
-        end
+      def start_element(name, _)
+        increment_depth(1) if depth_within_element > 0
```
this looks a lot better this way :+1:

soulcutter commented on spec/spec_helper.rb:
```diff
@@ -1,30 +1,33 @@
+
+  config.before(:suite) do |_|
+    unless adapter
+      puts 'SAXERATOR_ADAPTER weren\'t defined'
```
I don't think this should exit here, it should just skip this block. I think the `return` here probably should have been a `next`

soulcutter commented:
This caught some good stuff, both in style and substance.

One cop I would like to disable is the `StringLiterals` cop - policing single vs double quotes is a total waste, even in an automated tool.

fanantoxa commented:
@soulcutter I've updated rubocop config with your additions.
Also I've added to this PR next tasks:
[Drop support ruby 1.9.3](https://github.com/soulcutter/saxerator/issues/45)
[Add a changelog](https://github.com/soulcutter/saxerator/issues/41)

Please check when you find a time

soulcutter commented:
Looks great!
